common/metrics/src/lib.rs

```diff
@@ -90,8 +92,28 @@ fn get_metrics_file<P: AsRef<Path>>(dir_path: &P, file_name: &str) -> File {
         .expect("Open metrics file failed")
 }
 
+pub fn gather_metrics() -> Vec<prometheus::proto::MetricFamily> {
```

Comment:
Should we create a metric to track the total number of counters?

Reply:
We can do that.  The metric will just magically disappear however, when we start dropping metrics :D

Reply:
Done, added a metric for the number seen, and the number kept.  And purposely keep those metrics.